### PR TITLE
feat: add config formatting

### PR DESCRIPTION
**Pull request by shamsartem**

### CODE REVIEW

File: src/lib/configs/initConfig.ts [hunk 1]
```diff
@@ -438,6 +438,58 @@ export function getReadonlyConfigInitFunction<
...
+    .trim()
+    .split("\n")
+    .flatMap((line, i, ar) => {
+      // If it's empty string - it was a newline - remove it
+      if (line.trim() === "") {
+        return [];
+      }
+
+      const maybePreviousLine = ar[i - 1];
+
```
Comment (akim-bow) [inline src/lib/configs/initConfig.ts:452]: Can we check `undefined` once at a top of the function?

File: src/lib/configs/initConfig.ts [hunk 2]
```diff
@@ -438,6 +438,58 @@ export function getReadonlyConfigInitFunction<
...
+  return `${formattedConfig.trim()}\n`;
+}
+
+async function saveConfig(
+  configPath: string,
+  migratedConfigString: string,
+): Promise<string> {
+  const configToSave = formatConfig(migratedConfigString);
+  await writeFile(configPath, configToSave, FS_OPTIONS);
+  const savedConfig = configToSave;
```
Comment (akim-bow) [inline src/lib/configs/initConfig.ts:489]:
```suggestion
return configToSave;
```

Comment (akim-bow) [reply src/lib/configs/initConfig.ts:489]: I don't see a point to make another variable like this.
There is already variable named **configToSave** which stored config. It guess it's pretty obvious what's saving here.

Comment (shamsartem) [reply src/lib/configs/initConfig.ts:489]: Wanted to be a bit more explicit, but looks a little bit questionable, I can remove

File: src/lib/configs/initConfig.ts (hunk 1 above)
Comment (shamsartem) [reply src/lib/configs/initConfig.ts:454]: I reworked this part so no it's less comments - more veriable names instead and one less if statement - but the result is the same

**Review by akim-bow — APPROVED** [review]

**Pull request merged**